### ISSUE TITLE
Undo via `u` does not remove file dirty flag

### ISSUE THREAD

**Issue by jaredmcateer** [VSCodeVim/Vim#3201]
**Describe the bug**
Using `u` to undo changes doesn't restore the file to a clean state when reverting back to the original file that was opened.

**To Reproduce**
Steps to reproduce the behaviour:

1.  open file
2.  enter insert mode
2.  insert a space, observe tab has dirty icon
3.  exit to normal mode and press `u` to remove change
4.  File is still considered dirty

**Expected behaviour**
If you perform this same operation but instead of pressing `u` you press `ctrl`+`z` the file will return to a clean state, this should be the same for `u`

**Environment (please complete the following information):**

- Extension (VsCodeVim) version: 0.16.12
- VSCode version: 1.28.2
- OS: Ubuntu 18 LTS

**Additional context**
Related issue in vscode for context https://github.com/Microsoft/vscode/issues/919

**Comment by abbondanza** [comment]
This also applies to manual undos, not just those using `u`.

**Comment by abbondanza** [comment]
This is a quick solution to this problem (until there's an official one):
```
    "vim.normalModeKeyBindings": [
        {
            "before" : ["u"],
            "commands" : [
                "undo"
            ]
        },
        {
            "before" : ["<C-r>"],
            "commands" : [
                "redo"
            ]
        }
    ]
```

**Comment by Hultner** [comment]
> Here's a quick solution to this problem (until there's an official one):
> 
> ```
>     "vim.normalModeKeyBindings": [
>         {
>             "before" : ["u"],
>             "commands" : [
>                 "undo"
>             ]
>         },
>         {
>             "before" : ["<C-r>"],
>             "commands" : [
>                 "redo"
>             ]
>         }
>     ]
> ```

Neat trick, I came here for a different issue where undo seems to undo a ton of actions (much more then the editor undo) which makes undo/redo almost unusable for me at the moment.

**Comment by nzwane-r7** [comment]
> > Here's a quick solution to this problem (until there's an official one):
> > ```
> >     "vim.normalModeKeyBindings": [
> >         {
> >             "before" : ["u"],
> >             "commands" : [
> >                 "undo"
> >             ]
> >         },
> >         {
> >             "before" : ["<C-r>"],
> >             "commands" : [
> >                 "redo"
> >             ]
> >         }
> >     ]
> > ```
> 
> Neat trick, I came here for a different issue where undo seems to undo a ton of actions (much more then the editor undo) which makes undo/redo almost unusable for me at the moment.

@Hultner  this might be the issue related to your experience: https://github.com/VSCodeVim/Vim/issues/2007

**Comment by AitBits** [comment]
This issue seems like a duplicate of #1490 to me.

**Comment by jaredmcateer** [comment]
I disagree, all this ticket is asking is to set the file to pristine if the undo stack returns to the previous saved state. Now if the decision on that ticket is to rewrite the entire history stack to use vscode and this issue is resolved as a consequence then I'm fine with it being closed but otherwise I think they are currently two distinct issues

**Comment by J-Fields** [comment]
While it's not the same issue, I don't think this can be fixed with the current VSCode API until #1490 is fixed.

**Comment by MrHate** [comment]
> Here's a quick solution to this problem (until there's an official one):
> 
> ```
>     "vim.normalModeKeyBindings": [
>         {
>             "before" : ["u"],
>             "commands" : [
>                 "undo"
>             ]
>         },
>         {
>             "before" : ["<C-r>"],
>             "commands" : [
>                 "redo"
>             ]
>         }
>     ]
> ```

A confusing point is, after I use `O` to produce an indented new line, I need to press `u` twice to undo this operation. One for the indent, the other one for the new line.
